java/src/org/openqa/selenium/support/decorators/WebDriverDecorator.java

```diff
@@ -480,12 +488,12 @@ private static void extractInterfaces(final Set<Class<?>> collector, final Class
   private <Z> Map<Class<?>, Function<Z, InvocationHandler>> deriveAdditionalInterfaces(Z sample) {
     Map<Class<?>, Function<Z, InvocationHandler>> handlers = new HashMap<>();
 
-    if (sample instanceof WebDriver && !(sample instanceof WrapsDriver)) {
+    if (sample instanceof WebDriver) {
       handlers.put(
           WrapsDriver.class,
           (instance) ->
               (proxy, method, args) -> {
-                if ("getWrappedDriver".equals(method.getName())) {
+                if (sample instanceof WrapsDriver || "getWrappedDriver".equals(method.getName())) {
```

Comment:
Why should the method name be ignored for a WrapsDriver instance? WrapsDriver does only has this one method, but this might change over time. Or some generic methods e.g. `.hashCode()` could be called?

Reply:
@joerg1985 Yes, we want to ignore the method name.
Whatever method is called on wrapper, we want to call the same method on the underlying (wrapped) instance.